File: dart/lib/model/v3_direction_with_description.dart
part of swagger.api;

class V3DirectionWithDescription {
  
  String routeDirectionDescription = null;
/* Direction of travel identifier */
  int directionId = null;
/* Name of direction of travel */
  String directionName = null;
/* Route identifier */
  int routeId = null;
/* Transport mode identifier */
  int routeType = null;

  V3DirectionWithDescription();

  @override
  String toString() {
    return 'V3DirectionWithDescription[routeDirectionDescription=$routeDirectionDescription, directionId=$directionId, directionName=$directionName, routeId=$routeId, routeType=$routeType, ]';
  }

  V3DirectionWithDescription.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    routeDirectionDescription = json['route_direction_description'];
    directionId = json['direction_id'];
    directionName = json['direction_name'];
    routeId = json['route_id'];
    routeType = json['route_type'];
  }

  Map<String, dynamic> toJson() {
    return {
      'route_direction_description': routeDirectionDescription,
      'direction_id': directionId,
      'direction_name': directionName,
      'route_id': routeId,
      'route_type': routeType
     };
  }

  static List<V3DirectionWithDescription> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3DirectionWithDescription>() : json.map((value) => new V3DirectionWithDescription.fromJson(value)).toList();
  }

  static Map<String, V3DirectionWithDescription> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3DirectionWithDescription>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3DirectionWithDescription.fromJson(value));
    }
    return map;
  }
}

File: java-vertx/src/main/java/io/swagger/server/api/model/V3DirectionWithDescription.java
package io.swagger.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.vertx.core.json.JsonObject;
import io.vertx.codegen.annotations.DataObject;
/**
 * V3DirectionWithDescription
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaVertXServerCodegen", date = "2024-04-14T03:35:56.953023300+10:00[Australia/Sydney]")



public class V3DirectionWithDescription   {
  private String routeDirectionDescription = null;

  private Integer directionId = null;

  private String directionName = null;

  private Integer routeId = null;

  private Integer routeType = null;

  public V3DirectionWithDescription routeDirectionDescription(String routeDirectionDescription) {
    this.routeDirectionDescription = routeDirectionDescription;
    return this;
  }

  /**
   * Get routeDirectionDescription
   * @return routeDirectionDescription
   **/
    public String getRouteDirectionDescription() {
    return routeDirectionDescription;
  }

  public void setRouteDirectionDescription(String routeDirectionDescription) {
    this.routeDirectionDescription = routeDirectionDescription;
  }

  public V3DirectionWithDescription directionId(Integer directionId) {
    this.directionId = directionId;
    return this;
  }

  /**
   * Direction of travel identifier
   * @return directionId
   **/
    public Integer getDirectionId() {
    return directionId;
  }

  public void setDirectionId(Integer directionId) {
    this.directionId = directionId;
  }

  public V3DirectionWithDescription directionName(String directionName) {
    this.directionName = directionName;
    return this;
  }

  /**
   * Name of direction of travel
   * @return directionName
   **/
    public String getDirectionName() {
    return directionName;
  }

  public void setDirectionName(String directionName) {
    this.directionName = directionName;
  }

  public V3DirectionWithDescription routeId(Integer routeId) {
    this.routeId = routeId;
    return this;
  }

  /**
   * Route identifier
   * @return routeId
   **/
    public Integer getRouteId() {
    return routeId;
  }

  public void setRouteId(Integer routeId) {
    this.routeId = routeId;
  }

  public V3DirectionWithDescription routeType(Integer routeType) {
    this.routeType = routeType;
    return this;
  }

  /**
   * Transport mode identifier
   * @return routeType
   **/
    public Integer getRouteType() {
    return routeType;
  }

  public void setRouteType(Integer routeType) {
    this.routeType = routeType;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3DirectionWithDescription v3DirectionWithDescription = (V3DirectionWithDescription) o;
    return Objects.equals(this.routeDirectionDescription, v3DirectionWithDescription.routeDirectionDescription) &&
        Objects.equals(this.directionId, v3DirectionWithDescription.directionId) &&
        Objects.equals(this.directionName, v3DirectionWithDescription.directionName) &&
        Objects.equals(this.routeId, v3DirectionWithDescription.routeId) &&
        Objects.equals(this.routeType, v3DirectionWithDescription.routeType);
  }

  @Override
  public int hashCode() {
    return Objects.hash(routeDirectionDescription, directionId, directionName, routeId, routeType);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3DirectionWithDescription {\n");
    
    sb.append("    routeDirectionDescription: ").append(toIndentedString(routeDirectionDescription)).append("\n");
    sb.append("    directionId: ").append(toIndentedString(directionId)).append("\n");
    sb.append("    directionName: ").append(toIndentedString(directionName)).append("\n");
    sb.append("    routeId: ").append(toIndentedString(routeId)).append("\n");
    sb.append("    routeType: ").append(toIndentedString(routeType)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
